lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -168,10 +168,19 @@ WasmBytecodeGenerator::GenerateFunction()
 
     WasmOp op;
     EmitInfo exprInfo;
-    while ((op = m_reader->ReadExpr()) != wnLIMIT)
+
+    try
+    {
+        while ((op = m_reader->ReadExpr()) != wnLIMIT)
+        {
+            exprInfo = EmitExpr(op);
+            ReleaseLocation(&exprInfo);
+        }
+    }
+    catch (WasmCompilationException& e)
```

Comment:
Hmm, I try to keep exception handlers specific as possible to avoid unexpected behavior. I had considered making it a catch all. If the exception caught were from the ByteCodeWriter itself, would it be safe?